src/Hocon/Extensions/ArrayGetterExtensions.cs

```diff
@@ -0,0 +1,462 @@
+﻿// -----------------------------------------------------------------------
+// <copyright file="ArrayGetterExtensions.cs" company="Akka.NET Project">
+//      Copyright (C) 2013 - 2020 .NET Foundation <https://github.com/akkadotnet/hocon>
+// </copyright>
+// -----------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Hocon.Extensions
+{
+    public static class ArrayGetterExtensions
+    {
+        public static T[] ToArray<T>(this HoconElement element)
+        {
+            if (element is HoconObject obj)
+                return obj.ToArray().ToArray<T>();
+
+            if (!(element is HoconArray arr))
+                throw new HoconException(
+                    $"Array getter functions can only work on {nameof(HoconArray)} type. {element.GetType()} found instead.");
+
+            return arr.Select(v => v.Get<T>()).ToArray();
+        }
+
+        public static IList<HoconElement> GetArray(this HoconElement element)
+        {
+            if (element is HoconObject obj)
+                return obj.ToArray().ToList();
+
+            if (!(element is HoconArray arr))
+                throw new HoconException(
+                    $"Array getter functions can only work on {nameof(HoconArray)} type. {element.GetType()} found instead.");
+
+            return arr.ToList();
+        }
+
+        /// <summary>
+        ///     Retrieves a list of byte values from this <see cref="HoconElement" />.
+        /// </summary>
+        /// <returns>A list of byte values represented by this <see cref="HoconElement" />.</returns>
+        public static byte[] ToByteArray(this HoconElement element)
+        {
+            return element;
```

Comment:
How do we know that this object is already a `byte[]` type?

Reply:
We don't, the API will convert the data type on the fly as user requests it. The stored data does not enforce any data type contract, it is up to the user to interpret the data.
The conversion is done "automagically", see `GenericGetterExtensions.cs` and the `Casting operators` region in `HoconLiteral.cs`

Reply:
Basically, you can do `int a = hoconConfig["anInteger"]`, the code will detect the conversion/casting on its own and throws an exception if it fails.
So if the user tried to do `float f = hoconConfig["nonFloatString"]` the code will throw a conversion exception.